src/dlsproto/client/request/internal/GetRange.d

```diff
@@ -398,11 +398,49 @@ private scope class GetRangeHandler
 
     private /* scope */ class RecordStream
     {
-        /// The acquired buffer to store a batch of records.
-        private void[]* batch_buffer;
+        /// Double buffer structure
+        private struct DoubleBuffer
+        {
+            /// Acquired buffers to store batches of records. Reader always
+            /// appends to the back one and the RecordStream always uses the
+            /// front one.
+            private void[]*[2] batch_buffers;
+
+            /// Index of the batch we're currently iterating over
+            private int current_buffer;
```

Comment:
I would use `bool`...

Reply:
I would still prefer `int` though, because it doesn't restrict the number of the buffers artificially to two (it's easy to generalize this to N buffers this way). For the same reason I would stick with modulo arithmetic.

Reply:
As an alternative you could use `void[]* front_buffer, back_buffer` and swap them in the `swap` method. This would avoid the need for `current_buffer` and the buffer getter methods.

Reply:
OK, that's more neat. I'll take that suggestion :+1: